frontend/src/MatrixApi.js

```diff
@@ -27,6 +27,19 @@ export class MatrixApi {
         }
     }
 
+    requestOpenIdToken(type, baseUrl, data) {
```

Comment:
Much better, I would forgo the `type` parameter as I think it makes your logic quite complex (and the caller doesn't care about the HTTP method or path, they just want to execute an action and get some data back).

Instead, it's a better idea to have distinct functions when you have multiple HTTP methods e.g. GET /mything and POST /mything would be `getMyThing()` and `postMyThing()`. In this case though, I think you only have one valid method so `requestOpenIdToken` is fine.

Reply:
Done.